CONTRIBUTING.md

```diff
@@ -1,26 +1,137 @@
+The possible commits types prefixes are limited to those defined by [conventional-commit-types][commit_types].
+This promotes a clean project history and enabled automatically generating a changelog.
+
+The commit message format will be inspected both on a git pre-commit hook
+and during the central CI build and will **fail the build** if issues are found.
+
+It is recommended to use `git cz` to construct valid conventional commit messages.
 
-## Commit Messages
-This project uses [conventional commits standard](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#specification) with the [@commitlint/config-conventional](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
-Recommanded: Use `git cz` to build conventional commit messages.
 - requires [commitizen](https://github.com/commitizen/cz-cli#installing-the-command-line-tool) to be installed.
 
-## Developer Certificate of Origin (DCO)
-Due to legal reasons, contributors will be asked to accept a DCO before they submit the first pull request to this projects, this happens in an automated fashion during the submission process. SAP uses [the standard DCO text of the Linux Foundation](https://developercertificate.org/).
+[commit_types]: https://github.com/commitizen/conventional-commit-types/blob/master/index.json
+[conventional_commits]: https://www.conventionalcommits.org/en/v1.0.0/
+
+### Formatting.
+
+[Prettier](https://prettier.io/) is used to ensure consistent code formatting in this repository.
+This is normally transparent as it automatically activated in a pre-commit hook using [lint-staged](https://github.com/okonet/lint-staged).
+However, this does mean that dev flows that do not use a full dev env (e.g editing directly on github)
+may result in voter failures due to formatting errors.
+
+### Compiling
+
+Use the following npm scripts at the repo's **root** to compile **all** the TypeScript sub-packages.
+
+- `yarn compile`
+- `yarn compile:watch` (will watch files for changes and re-compile as needed)
+
+These scripts may also be available inside the sub-packages. However, it is recommended to
+use the top-level compilation scripts to avoid forgetting to (re-)compile a sub-package's dependency.
+
+#### Run the dev mode
+
+Dev mode allows you to run the framework in the browser, using vue cli for fast development cycles, and easy debug tools.
+To run it do the following:
+
+- In the packages/backend folder run `webpack` or `webpack-dev:watch`, then run the server.
+  ```bash
+  yarn webpack-dev:watch
+  yarn ws:run
+  ```
+- In the packages/frontend folder run `serve`
+  ```bash
+  yarn serve
+  ```
+- Open the broswer on `localhost:8080` to access the framework.
+
+#### Run the VSCode extension
+
+- Start VSCode on your local machine, and click on open workspace. Select this repo folder.
+- On the debug panel choose "Run Extension", and click on the "Run" button.
+
+#### Enable usage analytics reporting from VS Code
```

Comment:
this analytics section does not belong in the contributing guide.
probably in the README of the VSCode ext

Reply:
This is already in README of the VSCode ext. Removing from here. Thanks!